docs/content/docs/dev/table/materialized-table/overview.md

```diff
@@ -0,0 +1,150 @@
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Introduction
+----------------
+
+Materialized Table is a new feature introduced in Flink 1.20, aimed at simplifying the batch and stream processing development process, providing a consistent development experience. Users only need to specify the corresponding query and desired data freshness when creating a `Materialized Table`, and the framework will automatically select the appropriate execution mode to meet the data freshness requirements.
+
+{{< hint warning >}}
+This feature is in the MVP stage and is only supported for use with `SQL Gateway` and in `Flink standalone` mode. It is currently not recommended for production use.
+{{< /hint >}}
+
+# Core Concepts
+
+A `Materialized Table` mainly consists of three parts: refresh mode, data freshness, and query definition.
+
+## Refresh Mode
+
+The refresh mode determines how to update the `Materialized Table`. Flink supports `CONTINUOUS` mode and `FULL` mode by default. The refresh task in `CONTINUOUS` mode is a real-time job, while the refresh task in `FULL` mode is a batch job, which is persisted as a specific workflow task to an external scheduler.
+
+## Data Freshness
+
+Data freshness determines the refresh frequency of the data in the `Materialized Table`. Currently, Flink supports data update frequencies at different granularity such as seconds, minutes, hours, and days, but not months and years. In `CONTINUOUS` mode, data freshness is converted to the checkpoint interval of the background real-time job, while in `FULL` mode, data freshness is converted to the scheduling interval of the workflow.
+
+## Query Definition
+
+The query definition is used to automatically update the `Materialized Table`. To support switching between batch and stream for queries corresponding to the `Materialized Table`, the source tables used in the query need to support both batch and stream scan. For example, in `CONTINUOUS` mode, the source tables in the query will be regarded as **unbounded streams** so that the background Flink real-time job can continuously update the `Materialized Table`. In `FULL` mode, the source tables in the query will be regarded as **bounded streams**, and the background task will be a batch processing task.
+
+# Syntax
```

Comment:
Regarding the syntax explaination, we can refer to FLIP docs: https://cwiki.apache.org/confluence/display/FLINK/FLIP-435%3A+Introduce+a+New+Materialized+Table+for+Simplifying+Data+Pipelines